apps/examples/camera/camera_main.c

```diff
@@ -0,0 +1,382 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <errno.h>
+#include <debug.h>
+#include <fcntl.h>
+#include <video/video.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+#ifndef VIDEO_DEVNAME
+#define VIDEO_DEVNAME  "/dev/video%d"
+#endif
+/* Note: Buffer size must be multiple of 32. */
+
+#define IMAGE_JPG_SIZE     (512*1024)	/* 512kB */
```

Comment:
Instead of all these, we'd better make virtual driver like a audio_null.c (video_null)
Then we can bring format through the video_enum_fmt, where implemented in video_null.c.
I think video_null can use data from some contents in rom directory.

Reply:
Can you brief me a little more, i didnt understand why we need a null driver implementation here?

Reply:
I mean implement virtual driver is needed, then we can add video features to media framework, also video streaming service until uvc & isochronous completed.

Reply:
It is related to this patch, because application is just one, but many drivers.
So application should request to driver regarding format, to easy to this in this time, I mentioned virtual driver.
(Also virtual driver is useful when we start new chip board.)
If you want to do implement virtual driver later, then at least please add 'TODO' comment here, if you agree with my opinion regarding requesting.

Reply:
Will prepare an video null driver in another PR.